### ISSUE TITLE
prediction return only 3 output when training on multiple GPUs

### ISSUE THREAD

**Issue by Ori226** [fizyr/keras-retinanet#383]
I don't know why exactly this happens, but:
`
model.predict_on_batch(np.expand_dims(image, axis=0))
`
return only 3 output when training with multiple GPUs.
 
I think there are other issues in the multi-gpus mode ( I'm not sure but I think that the bounding box are in incorrect scale).

I can think of several ways to bypass it, but it looks to me like a bug.

**Comment by hgaiser** [comment]
Yes, multi-gpu is broken and buggy. That's why the `--multi-gpu-force` flag is added. I prefer to fix https://github.com/fizyr/keras-retinanet/issues/334 before fixing multi-gpu.

**Comment by jjiunlin** [comment]
I edit the model ouput to train with multiple gpu.
The prediction model is build on CPU so is very slow when evaluation.So I train with `--no-evaluation` 
 and run `evaluate_coco.py` by myself.
```
    if multi_gpu > 1:
        with tf.device('/cpu:0'):
            model = model_with_weights(backbone_retinanet(num_classes, backbone=backbone, nms=False, modifier=modifier), weights=weights, skip_mismatch=True)
            # append NMS for prediction only
            regression = model.outputs[0]
            classification = model.outputs[1]
            boxes = model.outputs[2]
            nms = layers.NonMaximumSuppression(name='nms')([boxes, classification])
            # construct list of outputs
            outputs = [regression, classification] + [boxes] + [nms]
            prediction_model = keras.models.Model(inputs=model.inputs, outputs=outputs)

        training_model = multi_gpu_model(model, gpus=multi_gpu)
```

**Comment by hgaiser** [comment]
Fixed with #386
